lib/musiclibrarycontroller.rb

```diff
@@ -48,14 +35,14 @@ def list_genres
   end
 
   def play_song
-    size = @control.files.length
+    size = Song.all.length
     Message.what_song(size)
-    index = gets.chomp.to_i
+    selection = gets.chomp.to_i - 1
 
-    if index < 1 || index > size
+    if [0..size].include?(selection) && selection
```

Comment:
This should not be accurate but it passed the tests. I found out the mistake I made was that I did not properly define the range (square brackets instead of round). It has been corrected and Message.invalid_song moved to the else part